### PR TITLE
Refactor NetconfConfigUtils - 14.2.x

### PR DESCRIPTION
**Pull request by martinbugan-pt**
Deprecate old methods in NetconfConfigUtils and provide a new ones for getting configuration and injecting services like in RestconfConfigUtils.

### CODE REVIEW

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java [hunk 1]
```diff
@@ -109,6 +109,62 @@ public static NetconfConfiguration createNetconfConfiguration(
...
+        return configuration;
+    }
+
+    /**
+     * Create default Netconf Southbound configuration from other NetconfConfiguration.
+     *
+     * @param configurationFrom other NetconfConfiguration to be created from.
+     * @return Object representation of configuration data.
+     */
+    public static NetconfConfiguration getNetconfConfiguration(final NetconfConfiguration configurationFrom) {
```
Comment (marekzatko) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:158]: I dont see the use of this method, is it suppose to copy config? I would delete this .. also lines 163 164 are wrong anyways.

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java [hunk 2]
```diff
@@ -118,21 +174,51 @@ public static NetconfConfiguration createDefaultNetconfConfiguration() {
...
+    public static NetconfConfiguration injectServicesToConfig(final NetconfConfiguration configuration)
+            throws ConfigurationException {
         final AAAEncryptionService aaa = NetconfConfigUtils.createAAAEncryptionService(
                 getDefaultAaaEncryptServiceConfig());
         configuration.setAaaService(aaa);
         return configuration;
     }
 
+    /**
+     * Inject services from LightyServices to Netconf southbound configuration.
```
Comment (marekzatko) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:209]: This does not inject them from LightyServices.. there is no LightyServices reference coming in. Also I think this method is pretty useless, user can do the same thing with one line.

File: lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java [hunk 3]
```diff
@@ -201,13 +304,27 @@ public static AAAEncryptionService createAAAEncryptionService(AaaEncryptServiceC
...
+                                                     final LightyServices services) {
         final NetconfClientDispatcher client =
                 new NetconfClientDispatcherImpl(services.getBossGroup(), services.getWorkerGroup(),
                         services.getTimer());
         configuration.setClientDispatcher(client);
         return configuration;
     }
 
+    /**
+     * Inject netconf client dispatcher to Netconf southbound configuration, it uses {@link LightyServices}.
```
Comment (marekzatko) [inline lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfConfigUtils.java:318]: It does not use LightyServices. I would also delete this method, it does nothing, user can achieve the same thing with one line.

**Review by marekzatko — CHANGES_REQUESTED** [review]

**Review by vainyksi — CHANGES_REQUESTED** [review]
The idea here should be to unify the config utils classes for NETCONF & RESTCONF.
I think we should specify in more detail what API the util classes should provide and for what the API methods are intended.

**Pull request closed without merging**